### PR TITLE
fix: unable to read wsl version when using chinese as syslang on Windows (#4863)

### LINKED ISSUE #4863: Unable to read the version of the WSLcorrectly
### Is your enhancement related to a problem? Please describe

When executing entire setup, the following error message is prompted
image

But my local WSL version is greater than required.
image

Can't recognize the WSL version of the machine, or is my WSL version incorrect?

### Describe the solution you'd like

Sorry, I couldn't find the feedback email, so I went to Github to create an issue. If need more detailed machine version information, I can provide it here

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR fixes an issue when detecting wsl version on a chinese lang system.
There were 2 problems, the first we encoded using utf8 when wsl returns an output encoded utf16, and then the output contains a fullwidth colon which is different from the one we use in ASCII.

### Screenshot/screencast of this PR

N/A 

### What issues does this PR fix or reference?

it fixes #4863

### How to test this PR?

1. switch to chinese on windows or just run on your usual syste mto see if it works as before

### CODE REVIEW

File: packages/main/src/plugin/util/exec.ts [hunk 1]
```diff
@@ -180,8 +180,8 @@ export class Exec {
         reject(errResult);
       });
 
-      childProcess.stdout.setEncoding('utf8');
-      childProcess.stderr.setEncoding('utf8');
+      childProcess.stdout.setEncoding(options?.encoding || 'utf8');
+      childProcess.stderr.setEncoding(options?.encoding || 'utf8');
```
Comment (benoitf) [inline packages/main/src/plugin/util/exec.ts]: it's minor but linter recommends nullish coalescing rather than the OR operator

Comment (benoitf) [reply packages/main/src/plugin/util/exec.ts]: `??`

**Review by benoitf — APPROVED** [review]
probably need to split in two, one part for extension api, encoding, etc, tests

and another PR for just podman

**Review by benoitf — COMMENTED** [review]
interesting reading and fix 👍

**Review by feloy — APPROVED** [review]
works on my Windows 10 machine

**Review by jeffmaury — APPROVED** [review]
LGTM. Maybe test should be enhanced to exhibit the difference values for :

**Comment by lstocchi** [comment]
just added a test for the small colon symbol so now all symbols are covered

**Pull request merged**